examples/vision/integrated_gradients.py

```diff
@@ -0,0 +1,474 @@
+"""
+Title: Integrated Gradients
+Author: [A_K_Nain](https://twitter.com/A_K_Nain)
+Date created: 2020/06/2
+Last modified: 2020/06/2
+Description: How to obtain integrated gradients for a classification model.
+"""
+
+"""
+## Integrated Gradients
+[Integrated Gradients](https://arxiv.org/abs/1701.07875) is a technique for
+attributing a classification model's prediction to its input features. It is
+one of the interpretibility techniques used to understand why did the model make
+this prediction on this input.
+
+[Integrated Gradients](https://arxiv.org/abs/1701.07875) involves following steps:
+1. Construct a sequence of images interpolating from a baseline to the actual image.
+2. Do a forward pass on these interpolated images and get the gradients.
+2. Average the gradients across these images.
+
+Refrences:
+Paper: https://arxiv.org/abs/1701.07875
+Original implementation: https://github.com/ankurtaly/Integrated-Gradients
+"""
+
+"""
+## Setup
+"""
+
+
+import numpy as np
+import matplotlib.pyplot as plt
+from scipy import ndimage
+from pathlib import Path
+from IPython.display import Image
+
+import tensorflow as tf
+from tensorflow import keras
+from tensorflow.keras import layers
+from tensorflow.keras.applications.xception import (
+    Xception,
+    decode_predictions,
+    preprocess_input,
+)
+
+# Size of the input image
+img_size = (299, 299, 3)
+
+# load xception model with imagenet weights
+model = Xception(weights="imagenet")
+
+# The local path to our target image
+img_path = keras.utils.get_file("elephant.jpg", "https://i.imgur.com/Bvro0YD.png")
+display(Image(img_path))
+
+"""
+## Integrated Gradients algorithm
+"""
+
+
+def get_img_array(img_path, size=(299, 299)):
+    # `img` is a PIL image of size 299x299
+    img = keras.preprocessing.image.load_img(img_path, target_size=size)
+    # `array` is a float32 Numpy array of shape (299, 299, 3)
+    array = keras.preprocessing.image.img_to_array(img)
+    return array
+
+
+def get_gradients(img_input, top_pred_idx):
+    """Computes the gradients of outputs w.r.t input image.
+
+    Args:
+        img_input: 4D image tensor
+        top_pred_idx: predicted label for the input image
+    Returns:
+        gradients of the predictions w.r.t to img_input
+    """
+    images = tf.cast(img_input, tf.float32)
+
+    # use gradient tape for computing gradients
```

Comment:
In general, capitalize all comments. This specific comment can be removed since it does not add more information compared to the code itself.